src/commands/init.ts

```diff
@@ -0,0 +1,116 @@
+import { Command, flags } from "@oclif/command";
+import { execSync } from "child_process";
+import * as fs from "fs";
+
+const DEFAULT_PROJECT_NAME = "react";
+
+/**
+ * @description a script that is used in the `package.json` scripts
+ * @example "g:component": "eb-scripts generate -t react-component -n"
+ */
+type Scripts = {
+  /* A script */
+  [script in string]: string;
+};
+
+/**
+ * @description valid project type
+ * @example `react`
+ * @todo eventually, we'll add others like `react-native`, `react-typescript`, `node`, etc.
+ */
+type ProjectType = "react";
+
+/**
+ * @description the available options for generate scripts that may be added to a user's `package.json`
+ */
+type ScriptByProject = {
+  [project in ProjectType]: Scripts;
+};
+
+// Grab the path of the user's project
+const pathWhereScriptIsRunning = process.cwd();
+const scriptsByProject: ScriptByProject = {
+  react: {
+    "g:component": "eb-scripts generate -t react-component -n"
+  }
+};
+export default class Init extends Command {
+  static description =
+    "initializes project by installing `eb-scripts` and adding scripts to `package.json`";
+
+  static examples = [`$ npx eb-scripts init -p react`];
+
+  static flags = {
+    help: flags.help({ char: "h" }),
+    // flag with a value (-p, --project=VALUE)
+    project: flags.string({
+      char: "p",
+      description: "language/framework of project"
+    })
+  };
+
+  static args = [
+    {
+      name: "project",
+      required: true,
+      description: "The language or framework of the project",
+      default: "react",
+      options: ["react"] as ProjectType[]
+    }
+  ];
+
+  async run() {
+    const { flags } = this.parse(Init);
+    const project = flags.project || DEFAULT_PROJECT_NAME;
+    // Grab their package Json
+    const packageJsonLocation = `${pathWhereScriptIsRunning}/package.json`;
+    const packageJson = require(packageJsonLocation);
+    const theirScripts = packageJson.scripts;
```

Comment:
might want to protect against a package without scripts. something like:

```suggestion
    const theirScripts = packageJson.scripts || {};
```